src/modules/fancyzones/dll/dllmain.cpp

```diff
@@ -195,11 +196,25 @@ class FancyZonesModule : public PowertoyModuleIface
     }
 
 private:
-    static bool IsInterestingWindow(HWND window)
+    bool IsInterestingWindow(HWND window)
     {
-        auto style = GetWindowLongPtr(window, GWL_STYLE);
-        auto exStyle = GetWindowLongPtr(window, GWL_EXSTYLE);
-        return IsWindowVisible(window) && WI_IsFlagSet(style, WS_MAXIMIZEBOX) && WI_IsFlagClear(style, WS_CHILD) && WI_IsFlagClear(exStyle, WS_EX_TOOLWINDOW);
+        auto windowAndPath = get_filtered_base_window_and_path(window);
+        if (windowAndPath.hwnd == nullptr)
+        {
+            return false;
+        }
+        CharUpperBuffW(windowAndPath.process_path.data(), (DWORD)windowAndPath.process_path.length());
+        if (m_settings)
+        {
+            for (const auto& excluded : m_settings->GetSettings().excludedAppsArray)
+            {
+                if (windowAndPath.process_path.find(excluded) != std::wstring::npos)

```

Comment:
This is a partial match, for example if the excluded app is "note" and the target app is notepad.exe, it will disable fancyzones for notepad.exe, this may generate confusion to the users.
We either force the users to add the application extension, like in "note.exe", or handle it in the code.

Reply:
Requiring an exact match would also be confusing, the app binary file name does not really have match exactly the app name. The best solution would probably be if a user can click on a window to disable it, maybe by using the system menu hook from #677?

Reply:
At least we need to document how it works, so a user may try to use the extension in order to avoid partial matching.
The FancyZones readme is outdated anyway and requires to be updated.
So let's open an issue for the Readme update after this PR is merged.